proofs/tree.go

```diff
@@ -204,17 +216,15 @@ func (doctree *DocumentTree) ValidateProof(proof *proofspb.Proof) (valid bool, e
 	return
 }
 
-// ValueToString takes any supported interface and returns a string representation of the value. This is used calculate
-// the hash and to create the proof object.
-func ValueToString(value interface{}) (s string, err error) {
+func DereferencePointer(value interface{}) (interface{}) {
```

Comment:
I think it is better to take `reflect.Value` as an argument and return `reflect.Value`